tiledb/bioimg/helpers.py

```diff
@@ -101,42 +113,71 @@ def create_image_pyramid(
     rw_group: ReadWriteGroup,
     base_uri: str,
     base_level: int,
-    max_tiles: Mapping[str, int],
+    max_tiles: MutableMapping[str, int],
     compressor: tiledb.Filter,
     pyramid_kwargs: Mapping[str, Any],
-) -> None:
-    with open_bioimg(base_uri) as a:
-        base_shape = a.shape
-        dim_names = tuple(dim.name for dim in a.domain)
-        dim_axes = "".join(dim_names)
-        attr_dtype = a.attr(0).dtype
-
-    scaler = Scaler(base_shape, dim_axes, **pyramid_kwargs)
+    base_shape: Tuple[int, ...],
+    preserve_axes: bool,
+    base_axes: Axes,
+    attr_dtype: np.dtype,
+) -> Sequence[Tuple[int, ...]]:
+    pixel_depth = get_pixel_depth(compressor)
+    if pixel_depth == 1:
+        if preserve_axes:
+            target_axes = base_axes
+        else:
+            target_axes = base_axes.canonical(base_shape)
+        axes_mapper = base_axes.mapper(target_axes)
+        dim_names = tuple(target_axes.dims)
+    else:
+        max_tiles["X"] *= pixel_depth
+        axes_mapper = base_axes.webp_mapper(pixel_depth)
+        dim_names = ("Y", "X")
+
+    scaler = Scaler(base_shape, base_axes.dims, **pyramid_kwargs)
```

Comment:
This is a code repetition from `_convert_level_to_tiledb`. Since reused consider making it a function.